skimage/morphology/_rolling_ball_cy.pyx

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+cimport cython
+from libc.math cimport isnan, INFINITY
+from cython.parallel cimport prange
+
+from .._shared.fused_numerics cimport np_floats
+
+ctypedef np_floats DTYPE_FLOAT
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def apply_kernel_nan(DTYPE_FLOAT[:,:,:] windows,
+                     DTYPE_FLOAT[:,::1] kernel,
+                     DTYPE_FLOAT[:,::1] cap_height,
+                     Py_ssize_t[::1] offsets):
+    """
+    apply_kernel_nan(windows, kernel, cap_height)
+
+    Apply a custom kernel to a windowed view of an image.
+
+    This function is the critical piece of code for 
+    `morphology.rolling_ellipsoid`. It was moved to cython for speed.
+
+    Parameters
+    ----------
+    windows : (N, M, K1, K2) ndarray
```

Comment:
This shape description (4 dim) is out of sync with the one in the function signature (3 dim)...?